src/zap/zapimage.cpp

```diff
@@ -1638,10 +1680,14 @@ void ZapImage::CompileProfileData()
                     {
                         m_pPreloader->AddMethodToTransitiveClosureOfInstantiations(pMethod);
 
-                        TryCompileInstantiatedMethod(pMethod, methodProfilingDataFlags);
+                        compileResult = TryCompileInstantiatedMethod(pMethod, methodProfilingDataFlags);
                     }
                 }
             }
+
+            // Update the 'flags' saved in the profileDataHashTable hash table.
+            //
+            hashBBUpdateFlagsAndCompileResult(token, methodProfilingDataFlags, compileResult);
         }
         // record the start of hot Generics methods.
         m_iGenericsMethod = m_MethodCompilationOrder.GetCount();
```

Comment:
I also took a look at that and it appear to me to be wrong and a left over from a time where we laid out all non-generic codes first followed by the generic code.  It feeds into a structure that records information about the structure of Ngen image. But I don't believe that the information has a consumer anymore, so it can be wrong without any impact. 
In any case we now lay out the code for generics in place along with the non-generic code, so that value isn't of any use to us anymore and could be removed.